tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_bw_and_latency.cpp

```diff
@@ -0,0 +1,162 @@
+}
+
+int main(int argc, char **argv) {
+    init(argc, argv);
+
+    bool pass = true;
+    try {
+        int device_id = 0;
+        tt_metal::Device *device = tt_metal::CreateDevice(device_id);
+
+        CommandQueue& cq = tt::tt_metal::detail::GetCommandQueue(device);
+
+        tt_metal::Program program = tt_metal::CreateProgram();
+
+        string src_mem;
+        switch (source_mem_g) {
+        case 0:
+            src_mem = "FROM_PCIE";
+            break;
+        case 1:
+            src_mem = "FROM_DRAM";
+            break;
+        case 2:
+            src_mem = "FROM_L1";
+            break;
+        }
+
+        std::map<string, string> defines = {
+            {src_mem, "1"},
+            {"ITERATIONS", std::to_string(iterations_g)},
+            {"PAGE_SIZE", std::to_string(page_size_g)},
+            {"PAGE_COUNT", std::to_string(page_count_g)},
+            {"LATENCY", std::to_string(latency_g)}
+        };
+
+        tt_metal::CircularBufferConfig cb_config = tt_metal::CircularBufferConfig(page_size_g * page_count_g, {{0, tt::DataFormat::Float32}})
+            .set_page_size(0, page_size_g);
+        auto cb = tt_metal::CreateCircularBuffer(program, worker_g, cb_config);
+
+        auto dm0 = tt_metal::CreateKernel(
+                                          program,
+                                          "tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/kernels/bw_and_latency.cpp",
+                                          worker_g,
+                                          tt_metal::DataMovementConfig{.processor = tt_metal::DataMovementProcessor::RISCV_0, .noc = tt_metal::NOC::RISCV_0_default, .defines = defines});
+
+        // Cache stuff
+        for (int i = 0; i < warmup_iterations_g; i++) {
+            EnqueueProgram(cq, program, false);
+        }
+        Finish(cq);
+
+        if (lazy_g) {
+            tt_metal::detail::SetLazyCommandQueueMode(true);
+        }
+
+        auto start = std::chrono::system_clock::now();
+        EnqueueProgram(cq, program, false);
+        if (time_just_finish_g) {
+            start = std::chrono::system_clock::now();
+        }
+        Finish(cq);
+        auto end = std::chrono::system_clock::now();
+
+        log_info(LogTest, "Reader core: {}", worker_g.start.str());
+        log_info(LogTest, "Reading: {}", src_mem);
+        log_info(LogTest, "Lazy: {}", lazy_g);
+        log_info(LogTest, "Size: {}", page_count_g * page_size_g);
+        log_info(LogTest, "Page size: {}", page_size_g);
+
+        std::chrono::duration<double> elapsed_seconds = (end-start);
+        log_info(LogTest, "Ran in {}us", elapsed_seconds.count() * 1000 * 1000);
+        if (latency_g) {
+            log_info(LogTest, "Latency: {} us", elapsed_seconds.count() / (page_count_g * iterations_g) * 1000.0 * 1000.0);
+        } else {
+            float bw = page_count_g * page_size_g * iterations_g / (elapsed_seconds.count() * 1000.0 * 1000.0 * 1000.0);
```

Comment:
Divided by 1024 cubed, not 1000 cubed

Reply:
actually, this is the time not the bytes, so we want 1000.  sorta odd, had this bug in a previous life (kilo-seconds isn't really a thing)